app/queries/content_item_uniqueness.rb

```diff
@@ -0,0 +1,57 @@
+module Queries
+  module ContentItemUniqueness
+    extend ArelHelpers
+    extend self
+
+    def unique_fields_for_content_item(content_item)
+      scope = unique_fields_scope
+      scope
+        .where(table(:content_items)[:id].eq(content_item.id))
+        .take(1)
+      get_rows(scope).first.try(:symbolize_keys)
+    end
+
+    def first_non_unique_item(content_item, base_path:, locale:, state:, user_facing_version:)
+      scope = unique_fields_scope
+      scope
+        .where(table(:content_items)[:id].not_eq(content_item.id))
+        .where(table(:states)[:name].eq(state))
+        .where(table(:translations)[:locale].eq(locale))
+        .where(table(:locations)[:base_path].eq(base_path))
+        .where(table(:user_facing_versions)[:number].eq(user_facing_version))
+        .take(1)
```

Comment:
the results are used in the error message at: https://github.com/alphagov/publishing-api/blob/bb069e68f21d8b47b0010ea8312e4924e26ad5d8/app/validators/content_item_uniqueness_validator.rb#L33